### PR TITLE
feat: fast-ish autofixable eslint config [WEB-1131]

### PR DESCRIPTION
**Pull request by jakemhiller**
### Overview

Added env var to disable type-aware linting for autofixing in commit hooks.

Added `eslint-fix` bin to eslint-config to make sharing the eslint autofixing config easier.

### PR Checklist

- [ ] Related to designs:
- [ ] Related to JIRA ticket: ABC-123
- [ ] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.d2991c.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.d2991c.0<br/>@codecademy/gamut-labs@7.13.5-alpha.d2991c.0<br/>@codecademy/gamut-styles@7.6.4-alpha.d2991c.0<br/>@codecademy/gamut-system@0.2.2-alpha.d2991c.0<br/>@codecademy/gamut@19.11.6-alpha.d2991c.0<br/>@codecademy/macros@1.1.3-alpha.d2991c.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.d2991c.0<br/>@codecademy/styleguide@21.20.3-alpha.d2991c.0<br/>@codecademy/webpack-config@5.2.2-alpha.d2991c.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6012f576dd01fa18b0bfb37a--gamut-preview.netlify.app>https://6012f576dd01fa18b0bfb37a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6012f576dd01fa18b0bfb37a>Deploy Logs</a></p>

File: .eslintrc.js [hunk 1]
```diff
@@ -21,20 +22,24 @@ module.exports = {
     },
     {
       files: ['**/typings/*', '*.d.ts'],
-      parserOptions: {
-        tsconfigRootDir: __dirname,
-        project: ['./tsconfig.json', './packages/*/tsconfig.json'],
-      },
+      parserOptions: TYPE_AWARE
```
Comment (JoshuaKGoldberg) [inline .eslintrc.js:25]: How about using a webpack-config style approach to building this with functions?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.3d58a1.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.3d58a1.0<br/>@codecademy/gamut-labs@7.13.5-alpha.3d58a1.0<br/>@codecademy/gamut-styles@7.6.4-alpha.3d58a1.0<br/>@codecademy/gamut-system@0.2.2-alpha.3d58a1.0<br/>@codecademy/gamut@19.11.6-alpha.3d58a1.0<br/>@codecademy/macros@1.1.3-alpha.3d58a1.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.3d58a1.0<br/>@codecademy/styleguide@21.20.3-alpha.3d58a1.0<br/>@codecademy/webpack-config@5.2.2-alpha.3d58a1.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6012fc76ce29f33e8b6b2e8f--gamut-preview.netlify.app>https://6012fc76ce29f33e8b6b2e8f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6012fc76ce29f33e8b6b2e8f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601309762013825f99784ee3--gamut-preview.netlify.app>https://601309762013825f99784ee3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601309762013825f99784ee3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.4-alpha.19b5f5.0<br/>@codecademy/gamut-illustrations@0.4.10-alpha.19b5f5.0<br/>@codecademy/gamut-labs@7.13.5-alpha.19b5f5.0<br/>@codecademy/gamut-styles@7.6.4-alpha.19b5f5.0<br/>@codecademy/gamut-system@0.2.2-alpha.19b5f5.0<br/>@codecademy/gamut@19.11.6-alpha.19b5f5.0<br/>@codecademy/macros@1.1.3-alpha.19b5f5.0<br/>@codecademy/markdown-overrides@0.4.4-alpha.19b5f5.0<br/>@codecademy/styleguide@21.20.3-alpha.19b5f5.0<br/>@codecademy/webpack-config@5.2.2-alpha.19b5f5.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.b2b66a.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.b2b66a.0<br/>@codecademy/gamut-labs@8.0.3-alpha.b2b66a.0<br/>@codecademy/gamut-styles@7.7.1-alpha.b2b66a.0<br/>@codecademy/gamut-system@0.2.2-alpha.b2b66a.0<br/>@codecademy/gamut@20.0.6-alpha.b2b66a.0<br/>@codecademy/macros@1.1.3-alpha.b2b66a.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.b2b66a.0<br/>@codecademy/styleguide@23.1.2-alpha.b2b66a.0<br/>@codecademy/webpack-config@5.2.2-alpha.b2b66a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601d68b7b9896f139c4c3ccb--gamut-preview.netlify.app>https://601d68b7b9896f139c4c3ccb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601d68b7b9896f139c4c3ccb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.a8fc79.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.a8fc79.0<br/>@codecademy/gamut-labs@8.0.3-alpha.a8fc79.0<br/>@codecademy/gamut-styles@7.7.1-alpha.a8fc79.0<br/>@codecademy/gamut-system@0.2.2-alpha.a8fc79.0<br/>@codecademy/gamut@20.0.6-alpha.a8fc79.0<br/>@codecademy/macros@1.1.3-alpha.a8fc79.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.a8fc79.0<br/>@codecademy/styleguide@23.1.2-alpha.a8fc79.0<br/>@codecademy/webpack-config@5.2.2-alpha.a8fc79.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601d6dd2bcf6cd13ec01beab--gamut-preview.netlify.app>https://601d6dd2bcf6cd13ec01beab--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601d6dd2bcf6cd13ec01beab>Deploy Logs</a></p>

File: .eslintrc.js (hunk 1 above)
Comment (jakemhiller) [reply .eslintrc.js]: it seems like a good idea, but `webpack-config` has been such a pain to maintain. I extracted this specifically out into a function which feels a bit cleaner.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.d4c98b.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.d4c98b.0<br/>@codecademy/gamut-labs@8.0.3-alpha.d4c98b.0<br/>@codecademy/gamut-styles@7.7.1-alpha.d4c98b.0<br/>@codecademy/gamut-system@0.2.2-alpha.d4c98b.0<br/>@codecademy/gamut@20.0.6-alpha.d4c98b.0<br/>@codecademy/macros@1.1.3-alpha.d4c98b.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.d4c98b.0<br/>@codecademy/styleguide@23.1.2-alpha.d4c98b.0<br/>@codecademy/webpack-config@5.2.2-alpha.d4c98b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601d70753966b2059106be79--gamut-preview.netlify.app>https://601d70753966b2059106be79--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601d70753966b2059106be79>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.f23a71.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.f23a71.0<br/>@codecademy/gamut-labs@8.0.3-alpha.f23a71.0<br/>@codecademy/gamut-styles@7.7.1-alpha.f23a71.0<br/>@codecademy/gamut-system@0.2.2-alpha.f23a71.0<br/>@codecademy/gamut@20.0.6-alpha.f23a71.0<br/>@codecademy/macros@1.1.3-alpha.f23a71.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.f23a71.0<br/>@codecademy/styleguide@23.1.2-alpha.f23a71.0<br/>@codecademy/webpack-config@5.2.2-alpha.f23a71.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601d731ba96f4c3eaf17cb2d--gamut-preview.netlify.app>https://601d731ba96f4c3eaf17cb2d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601d731ba96f4c3eaf17cb2d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.cc96fd.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.cc96fd.0<br/>@codecademy/gamut-labs@8.0.3-alpha.cc96fd.0<br/>@codecademy/gamut-styles@7.7.1-alpha.cc96fd.0<br/>@codecademy/gamut-system@0.2.2-alpha.cc96fd.0<br/>@codecademy/gamut@20.0.6-alpha.cc96fd.0<br/>@codecademy/macros@1.1.3-alpha.cc96fd.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.cc96fd.0<br/>@codecademy/styleguide@23.1.2-alpha.cc96fd.0<br/>@codecademy/webpack-config@5.2.2-alpha.cc96fd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://601dc93ba96f4ca39e17c427--gamut-preview.netlify.app>https://601dc93ba96f4ca39e17c427--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/601dc93ba96f4ca39e17c427>Deploy Logs</a></p>

File: .eslintrc.fix.js [hunk 2]
```diff
@@ -0,0 +1 @@
+module.exports = require('./packages/eslint-config');
```
Comment (jakemhiller) [inline .eslintrc.fix.js:1]: delete this

File: package.json [hunk 3]
```diff
@@ -30,6 +30,9 @@
   "lint-staged": {
     "**/*.{mdx,js,ts,tsx,json,css,scss}": [
       "prettier --ignore-path .prettierignore --write"
+    ],
+    "**/*.{mdx,js,ts,tsx,json}": [
+      "packages/eslint-config/bin/eslint-fix.js"
```
Comment (jakemhiller) [inline package.json:35]: combine these so the order is correct (eslint then prettier)

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.e149f9.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.e149f9.0<br/>@codecademy/gamut-labs@8.0.3-alpha.e149f9.0<br/>@codecademy/gamut-styles@7.7.1-alpha.e149f9.0<br/>@codecademy/gamut-system@0.2.2-alpha.e149f9.0<br/>@codecademy/gamut@20.0.6-alpha.e149f9.0<br/>@codecademy/macros@1.1.3-alpha.e149f9.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.e149f9.0<br/>@codecademy/styleguide@23.1.2-alpha.e149f9.0<br/>@codecademy/webpack-config@5.2.2-alpha.e149f9.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60216b5a83c314a9091bdb98--gamut-preview.netlify.app>https://60216b5a83c314a9091bdb98--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60216b5a83c314a9091bdb98>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.be9700.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.be9700.0<br/>@codecademy/gamut-labs@8.0.3-alpha.be9700.0<br/>@codecademy/gamut-styles@7.7.1-alpha.be9700.0<br/>@codecademy/gamut-system@0.2.2-alpha.be9700.0<br/>@codecademy/gamut@20.0.6-alpha.be9700.0<br/>@codecademy/macros@1.1.3-alpha.be9700.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.be9700.0<br/>@codecademy/styleguide@23.1.2-alpha.be9700.0<br/>@codecademy/webpack-config@5.2.2-alpha.be9700.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6021704f095b32c4b39e40e9--gamut-preview.netlify.app>https://6021704f095b32c4b39e40e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6021704f095b32c4b39e40e9>Deploy Logs</a></p>

File: .eslintrc.js [hunk 4]
```diff
@@ -1,24 +1,9 @@
...
- * move to a shared private module instead of having this in the
- * project root
- */
-const defaultConfig = require('./packages/eslint-config');
-
 module.exports = {
-  ...defaultConfig,
   root: true,
 
+  extends: [require.resolve('./packages/eslint-config')],
```
Comment (jakemhiller) [inline .eslintrc.js:4]: this works now 🎊

File: packages/eslint-config/index.js [hunk 5]
```diff
@@ -1,3 +1,5 @@
+const localExtends = ['./rules/typescript', './rules/mdx'].map(require.resolve);
```
Comment (jakemhiller) [inline packages/eslint-config/index.js:1]: stole this folder pattern from airbnb, might be nice to extend it to some of the other rules areas in this file. In this case i just split out the overrides.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.9dcbf5.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.9dcbf5.0<br/>@codecademy/gamut-labs@8.0.3-alpha.9dcbf5.0<br/>@codecademy/gamut-styles@7.7.1-alpha.9dcbf5.0<br/>@codecademy/gamut-system@0.2.2-alpha.9dcbf5.0<br/>@codecademy/gamut@20.0.6-alpha.9dcbf5.0<br/>@codecademy/macros@1.1.3-alpha.9dcbf5.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.9dcbf5.0<br/>@codecademy/styleguide@23.1.2-alpha.9dcbf5.0<br/>@codecademy/webpack-config@5.2.2-alpha.9dcbf5.0<br/></pre>

File: packages/eslint-config/bin/eslint-fix.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env node
+const path = require('path');
+const yargs = require('yargs/yargs');
+const { hideBin } = require('yargs/helpers');
```
Comment (JoshuaKGoldberg) [inline packages/eslint-config/bin/eslint-fix.js:4]: Not blocking at all, just a curiosity: does `require('yargs')` not work?

File: packages/eslint-config/bin/eslint-fix.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+const { argv } = yargs(hideBin(process.argv))
+  .options('config', {
+    alias: 'c',
+    description: 'path you your ESLint config',
```
Comment (JoshuaKGoldberg) [inline packages/eslint-config/bin/eslint-fix.js:9]:
```suggestion
    description: 'path to your ESLint config',
```

File: packages/eslint-config/rules/mdx.js [hunk 8]
```diff
@@ -0,0 +1,16 @@
+const mdxConfig = {
+  plugins: ['unused-imports'],
+  files: ['*.mdx'],
+  parser: 'eslint-mdx',
+  // Add this for MDX specifically since we rely on TS in other
```
Comment (JoshuaKGoldberg) [inline packages/eslint-config/rules/mdx.js:5]: in other...?

**Review by JoshuaKGoldberg — APPROVED** [review]
This looks & feels great. Very nice!

File: packages/eslint-config/rules/mdx.js (hunk 8 above)
Comment (jakemhiller) [reply packages/eslint-config/rules/mdx.js:5]: copy/paste from the existing config, i'll just remove it

Comment (jakemhiller) [reply packages/eslint-config/rules/mdx.js:5]:
```suggestion
```

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/eslint-config@3.1.7-alpha.3e913e.0<br/>@codecademy/gamut-illustrations@0.4.19-alpha.3e913e.0<br/>@codecademy/gamut-labs@8.0.3-alpha.3e913e.0<br/>@codecademy/gamut-styles@7.7.1-alpha.3e913e.0<br/>@codecademy/gamut-system@0.2.2-alpha.3e913e.0<br/>@codecademy/gamut@20.0.6-alpha.3e913e.0<br/>@codecademy/macros@1.1.3-alpha.3e913e.0<br/>@codecademy/markdown-overrides@0.4.5-alpha.3e913e.0<br/>@codecademy/styleguide@23.1.2-alpha.3e913e.0<br/>@codecademy/webpack-config@5.2.2-alpha.3e913e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60217ba749428c895b600f10--gamut-preview.netlify.app>https://60217ba749428c895b600f10--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60217ba749428c895b600f10>Deploy Logs</a></p>

**Pull request merged**